firmware/Makefile

```diff
@@ -14,18 +12,9 @@ clean: FORCE
 	rm $(BUILD_DIR) -rf
 
 deepclean: clean
-	@echo "Deleting files and directories in $(CUBEMX_DIR) except $(KEEP_FILES)"
-	@cd $(CUBEMX_DIR) && for file in *; do \
-		if ! echo "$(KEEP_FILES)" | grep -q "\\b$$file\\b"; then \
-			if [ -d "$$file" ]; then \
-				echo "Deleting directory $$file"; \
-				rm -rf "$$file"; \
-			else \
-				echo "Deleting file $$file"; \
-				rm -f "$$file"; \
-			fi; \
-		fi; \
-	done
+	@echo "Deleting generated files and directories in $(CUBEMX_DIR) that are ignored by Git"
+	@cd $(CUBEMX_DIR) && \
+		git check-ignore * | xargs -I{} rm -rf "{}"
```

Comment:
looks good, i like the use of a pipe.

Reply:
I just realized that since `git check-ignore *` only searches the current directory, it will not report folders that have even 1 non-ignored file.

the `find` command will recursively report all files and subdirectories, allowing `git check-ignore` to individually assess each

```console
find | xargs git check-ignore | xargs rm -rf
```

also notice that the default behaviour of `xargs` is to places the contents after the command, so you do not need the `-I{}` replacement syntax

Reply:
Is there a reason for the `&& \`? IIRC this executes the two commands in sequence, but can't this be achieved by simply writing the `git check-ignore` command on the next line?

Reply:
> Is there a reason for the `&& \`? IIRC this executes the two commands in sequence, but can't this be achieved by simply writing the `git check-ignore` command on the next line?

Yeah its the same thing I initially wanted the command to be one line and just kept it as it is.